vectordb_bench/backend/clients/memorydb/memorydb.py

```diff
@@ -0,0 +1,256 @@
+from .config import MemoryDBIndexConfig
+import redis
+from redis import Redis
+from redis.cluster import RedisCluster
+from redis.commands.search.field import TagField, VectorField, NumericField
+from redis.commands.search.indexDefinition import IndexDefinition, IndexType
+from redis.commands.search.query import Query
+import numpy as np
+
+
+log = logging.getLogger(__name__)
+INDEX_NAME = "index"    # Vector Index Name
+
+class MemoryDB(VectorDB):
+    def __init__(
+            self,
+            dim: int,
+            db_config: dict,
+            db_case_config: MemoryDBIndexConfig,
+            drop_old: bool = False,
+            **kwargs
+        ):
+
+        self.db_config = db_config
+        self.case_config = db_case_config
+        self.collection_name = INDEX_NAME
+        self.target_nodes = RedisCluster.RANDOM if not self.db_config["cmd"] else None
+        self.insert_batch_size = db_case_config.insert_batch_size or 1
+        self.dbsize = kwargs.get("num_rows")
+
+        # Create a redis connection, if db has password configured, add it to the connection here and in init():
+        log.info(f"Redis establishing connection to: {self.db_config}")
+        conn = self.get_client(primary=True)
+        log.info(f"Connection established: {conn}")
+        log.info(conn.execute_command("INFO server"))
+
+        if drop_old:
+            try:
+                log.info(f"Redis client getting info for: {INDEX_NAME}")
+                info = conn.ft(INDEX_NAME).info()
+                log.info(f"Index info: {info}")
+            except redis.exceptions.ResponseError as e:
+                log.error(e)
+                drop_old = False
+                log.info(f"Redis client drop_old collection: {self.collection_name}")
+            
+            log.info("Executing FLUSHALL")
+            conn.flushall()
+
+            # Since the default behaviour of FLUSHALL is asynchronous, wait for db to be empty
+            self.wait_until(self.wait_for_empty_db, 3, "", conn)
+            if not self.db_config["cmd"]:
+                replica_clients = self.get_client(replicas=True)
+                for rc, host in replica_clients:
+                    self.wait_until(self.wait_for_empty_db, 3, "", rc)
+                    log.debug(f"Flushall done in the host: {host}")
+                    rc.close()
+        
+        self.make_index(dim, conn)
+        conn.close()
+        conn = None
+
+    def make_index(self, vector_dimensions: int, conn: redis.Redis):
+        try:
+            # check to see if index exists
+            conn.ft(INDEX_NAME).info()
+        except Exception as e:
+            log.warn(f"Error getting info for index '{INDEX_NAME}': {e}")
+            index_param = self.case_config.index_param()
+            search_param = self.case_config.search_param()
+            vector_parameters = {  # Vector Index Type: FLAT or HNSW
+                "TYPE": "FLOAT32",  # FLOAT32 or FLOAT64
```

Comment:
Is the index type supposed to be configurable? If so, we should allow for this to be passed down as an option.

Reply:
Currently, the only supported type is `FLOAT32`. Will remove `FLOAT64` from the comments. 
We have really tested FLAT performance yet. We can make it configurable in future enhancements